src/gui/macOS/fileproviderdomainmanager_mac.mm

```diff
@@ -0,0 +1,660 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "configfile.h"
+#import <FileProvider/FileProvider.h>
+
+#include <QLoggingCategory>
+
+#include "config.h"
+#include "fileproviderdomainmanager.h"
+#include "pushnotifications.h"
+
+#include "gui/accountmanager.h"
+#include "libsync/account.h"
+
+// Ensure that conversion to/from domain identifiers and display names
+// are consistent throughout these classes
+namespace {
+
+QString domainIdentifierForAccount(const OCC::Account * const account)
+{
+    Q_ASSERT(account);
+    return account->userIdAtHostWithPort();
+}
+
+QString domainIdentifierForAccount(const OCC::AccountPtr account)
+{
+    return domainIdentifierForAccount(account.get());
+}
+
+QString domainDisplayNameForAccount(const OCC::Account * const account)
```

Comment:
why not `OCC::AccountPtr` ?

Reply:
We do have one with `OCC::AccountPtr` but for convenience, we can just overload rather than having to do the conversion each time depending on what we receive

Reply:
Because this is what we receive from the Account/PushNotification signals:

Pushnotifications:
```
void filesChanged(OCC::Account *account);
```

Account:
```
    void pushNotificationsReady(OCC::Account *account);

```

Reply:
I would say this is out of scope as it affects some other parts of the client -- I'll open a new PR with these changes